src/test/java/core/basesyntax/service/impl/FruitTransactionParserImplTest.java

```diff
@@ -0,0 +1,152 @@
+package core.basesyntax.service.impl;
+
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+
+class FruitTransactionParserImplTest {
+    private static final String HEADER = "type;fruit;quantity";
+    private final FruitTransactionParserImpl fruitTransactionParser =
+            new FruitTransactionParserImpl();
+    private final FruitTransaction fruitTransaction1 = new FruitTransaction(FruitTransaction
+            .Operation.BALANCE, "banana", 50);
+    private final String readLine1 = "b;banana;50";
+    private final FruitTransaction fruitTransaction2 = new FruitTransaction(FruitTransaction
+            .Operation.BALANCE, "apple", 100);
+    private final String readLine2 = "b;apple;100";
+    private final FruitTransaction fruitTransaction3 = new FruitTransaction(FruitTransaction
+            .Operation.BALANCE, "orange", 150);
+    private final String readLine3 = "b;orange;150";
+    private final FruitTransaction fruitTransaction4 = new FruitTransaction(FruitTransaction
+            .Operation.SUPPLY, "banana", 30);
+    private final String readLine4 = "s;banana;30";
+    private final FruitTransaction fruitTransaction5 = new FruitTransaction(FruitTransaction
```

Comment:
```
List<String> testData = new ArrayList<>(Arrays.asList(
                "type,fruit,quantity",
                "b,banana,20",
                "s,banana,100",
                "p,banana,13",
                "r,apple,10"));
        List<FruitTransaction> result = parserService.parseTransactions(testData);

FruitTransaction transaction1 = result.get(0);
        assertEquals(FruitTransaction.Operation.BALANCE,transaction1.getOperation());
        assertEquals("banana",transaction1.getFruit());
        assertEquals(20, transaction1.getQuantity());
```
        
You can do smth like that

Reply:
Nice ides at all. Assuming that such comment "can do" is not obligatory and means another approach for writing this test, could you approve this test with my approach. I've also discussed this issue in chat.